tests/csm/rest/test_information.py

```diff
@@ -0,0 +1,211 @@
+    @pytest.mark.parallel
+    @pytest.mark.cluster_user_ops
+    @pytest.mark.tags('TEST-42719')
+    def test_42719(self):
+        """
+        Test that user can check version compatability for invalid payload.
+        """
+        test_case_name = cortxlogging.get_frame()
+        self.log.info("##### Test started -  %s #####", test_case_name)
+
+        test_cfg = self.csm_conf["test_42719"]
+        resp_error_code = test_cfg["error_code"]
+        resp_msg_id = test_cfg["message_id"]
+        resp_data = self.rest_resp_conf[resp_error_code][resp_msg_id]
+        resp_msg_index = test_cfg["message_index"]
+        msg = resp_data[resp_msg_index]
+
+        self.log.info("[START] Testing Version Compatability with invalid rules")
+        payload = self.csm_obj.get_version_compatibility_payload("invalid_rules")
+        self.log.info("payload :  %s", payload)
+        response = self.csm_obj.verify_version_compatibility("node", "cortx-cluster", payload)
+        assert response.status_code == HTTPStatus.BAD_REQUEST, "Status code check failed"
+
+        if CSM_REST_CFG["msg_check"] == "enable":
+            self.log.info("Verifying error response...")
+            assert_utils.assert_equals(response.json()["error_code"], str(resp_error_code))
+            assert_utils.assert_equals(response.json()["message_id"], resp_msg_id)
+            assert_utils.assert_equals(response.json()["message"],
+                                        Template(msg).substitute(rule=payload["requires"]))
+
+        self.log.info("[END] Testing Version Compatability  with invalid rules")
+        self.log.info("##### Test ended -  %s #####", test_case_name)
+
+    @pytest.mark.lc92
+    @pytest.mark.parallel
+    @pytest.mark.cluster_user_ops
+    @pytest.mark.tags('TEST-42792')
+    def test_42792(self):
+        """
+        Test that user can check version compatability for invalid request body.
+        """
+        test_case_name = cortxlogging.get_frame()
+        self.log.info("##### Test started -  %s #####", test_case_name)
+
+        test_cfg = self.csm_conf["test_42792"]
+        resp_error_code = test_cfg["error_code"]
+        resp_msg_id = test_cfg["message_id"]
+        resp_data = self.rest_resp_conf[resp_error_code][resp_msg_id]
+        resp_msg_index = test_cfg["message_index"]
+        msg = resp_data[resp_msg_index]
+
+        self.log.info("[START] Testing Version Compatability with invalid request body")
+        payload = self.csm_obj.get_version_compatibility_payload("invalid_unknown_field")
+        self.log.info("payload :  %s", payload)
+        response = self.csm_obj.verify_version_compatibility("node", "cortx-cluster", payload)
+        assert response.status_code == HTTPStatus.BAD_REQUEST, "Status code check failed"
+
+        if CSM_REST_CFG["msg_check"] == "enable":
+            self.log.info("Verifying error response...")
+            assert_utils.assert_equals(response.json()["error_code"], str(resp_error_code))
```

Comment:
resp_error_code should not be string
[CORTX-31296](https://jts.seagate.com/browse/CORTX-31296)

Reply:
updated